lib/auth/auth.js

```diff
@@ -38,20 +38,33 @@ auth.check = (request, log, awsService, data) => {
         // handle temporary security credentials
         if (authHeader.startsWith('AWS ')) {
             log.trace('authenticating request with auth v2 using headers');
-            return authV2.headerAuthCheck.check(request, log, data);
+            const res = authV2.headerAuthCheck.check(request, log, data);
+            res.authType = 'REST-HEADER';
```

Comment:
I think those information can be into headerAuthCheck :) Same for the three others :)